monitoring/cgroup.go

```diff
@@ -46,35 +46,47 @@ type cgroupChecker struct {
 
 // Name returns name of the checker
 // Implements health.Checker
-func (r cgroupChecker) Name() string {
+func (c *cgroupChecker) Name() string {
 	return cgroupCheckerID
 }
 
-// Check verifies existence of cgroup mounts given in r.cgroups.
+// Check verifies existence of cgroup mounts given in c.cgroups.
 // Implements health.Checker
-func (r cgroupChecker) Check(ctx context.Context, reporter health.Reporter) {
-	var probes health.Probes
-	err := r.check(ctx, &probes)
-	if err != nil && !trace.IsNotFound(err) {
-		reporter.Add(NewProbeFromErr(r.Name(), "failed to validate cgroup mounts", err))
-		return
+func (c *cgroupChecker) Check(ctx context.Context, reporter health.Reporter) {
+	probesCh := make(chan health.Reporter, 1)
+	go func() {
+		probes, err := c.check()
```

Comment:
Here, again, the point is to cancel or at least minimize the actual work done, so shifting context inside of `check` make more sense. This needs to know how cancel the `consistentRead` operation:
 * first, it uses attempts so a basic `ctx.Done()` check inside a loop must take place
 * it uses `ioutil.ReadFile` for reading - could instead be `ioutil.ReadAll` with a reader that is context-aware.